File: Codeforces/54 Beta Division 2/Problem A/A.cc
// Chat room
#include <bits/stdc++.h>

using namespace std;

int main(){
	string a, b;
	a = "hello";
	cin >> b;
	
	int j = 0;
	for(int i = 0; i < b.length(); i++){
		if(a[j] == b[i]) j++;
		if(j == a.length()){
			puts("YES");
			return 0;
		}
	}
	
	puts("NO");
	return 0;
}

File: Codeforces/54 Beta Division 2/Problem A/A.py
s = raw_input()
l = len(s)
if 'h' in s:
	i = s.index('h')
	s = s[i+1:]
	if 'e' in s:
		i = s.index('e')
		s = s[i+1:]
		if 'l' in s:
			i = s.index('l')
			s = s[i+1:]
			if 'l' in s:
				i = s.index('l')
				s = s[i+1:]
				if 'o' in s:
					print "YES"
				else:
					print "NO"
			else:
				print "NO"
		else:
			print "NO"
	else:
		print "NO"
else:
	print "NO"
